consensus/istanbul/backend/announce.go

```diff
@@ -1067,27 +1149,39 @@ func (sb *Backend) handleEnodeCertificateMsg(peer consensus.Peer, payload []byte
 		return err
 	}
 
-	isFromProxiedPeer := sb.config.Proxy && sb.proxiedPeer != nil && sb.proxiedPeer.Node().ID() == peer.Node().ID()
-
-	// Handle the special case where this node is a proxy and the proxied validator sent the msg
-	if isFromProxiedPeer && msg.Address == sb.config.ProxiedValidatorAddress {
-		existingVersion := sb.getEnodeCertificateMsgVersion()
-		if enodeCertificate.Version < existingVersion {
-			logger.Warn("Enode certificate from proxied peer contains version lower than existing enode msg", "msg version", enodeCertificate.Version, "existing", existingVersion)
-			return errors.New("Version too low")
-		}
-		// There may be a difference in the URLv4 string because of `discport`,
-		// so instead compare the ID
-		selfNode := sb.p2pserver.Self()
-		if parsedNode.ID() != selfNode.ID() {
-			logger.Warn("Received Istanbul Enode Certificate message with an incorrect enode url", "message enode url", enodeCertificate.EnodeURL, "self enode url", sb.p2pserver.Self().URLv4())
-			return errors.New("Incorrect enode url")
-		}
-		if err := sb.setEnodeCertificateMsg(&msg); err != nil {
-			logger.Warn("Error setting enode certificate msg", "err", err)
-			return err
+	if sb.config.Proxy && sb.proxiedPeer != nil {
+		if sb.proxiedPeer.Node().ID() == peer.Node().ID() {
+			// if this message is from the proxied peer and contains the proxied
+			// validator's enodeCertificate, save it for handshake use
+			if msg.Address == sb.config.ProxiedValidatorAddress {
+				existingVersion := sb.getEnodeCertificateMsgVersion()
+				if enodeCertificate.Version < existingVersion {
+					logger.Warn("Enode certificate from proxied peer contains version lower than existing enode msg", "msg version", enodeCertificate.Version, "existing", existingVersion)
+					return errors.New("Version too low")
+				}
+				// There may be a difference in the URLv4 string because of `discport`,
+				// so instead compare the ID
+				selfNode := sb.p2pserver.Self()
+				if parsedNode.ID() != selfNode.ID() {
+					logger.Warn("Received Istanbul Enode Certificate message with an incorrect enode url", "message enode url", enodeCertificate.EnodeURL, "self enode url", sb.p2pserver.Self().URLv4())
+					return errors.New("Incorrect enode url")
+				}
+				if err := sb.setEnodeCertificateMsg(&msg); err != nil {
+					logger.Warn("Error setting enode certificate msg", "err", err)
+					return err
+				}
+				return nil
+			}
+		} else {
+			// If this message is not from the proxied validator, send it to the
+			// proxied validator without upserting it in this node. If the validator
+			// decides this proxy should upsert the enodeCertificate, then it
+			// will send it back to this node.
+			if err := sb.sendEnodeCertificateMsg(sb.proxiedPeer, &msg); err != nil {
+				logger.Warn("Error forwarding enodeCertificate to proxied validator", "err", err)
```

Comment:
I would argue this should be raised to an error because the result of a failure to send this message is that this proxy will not consider the peer a validator.
```suggestion
				logger.Error("Error forwarding enodeCertificate to proxied validator", "err", err)
```